cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -34,15 +36,23 @@
 
 class EntryPoint(Input, ClickableMixin):
     def fill(self, value):
-        if super(EntryPoint, self).fill(value):
-            self.parent_view.modal.cancel.click()
-            # After changing default path values of 'retirement_entry_point',
-            # 'reconfigure_entry_point' or 'provisioning_entry_point';
-            # 'Select Entry Point Instance'(pop up) occures which also helps to select these paths
-            # using tree structure. Here we have already filled these values through text input.
-            # So to ignore this pop up clicking on cancel button is required.
-            return True
-        return False
+        for _ in range(2):
+            try:
```

Comment:
If possible, I would try to avoid this structure. widgetastic itself already has checks for `StaleElementReferenceException`. Does the exception occur during the fill or when you try to click `cancel`? If the latter, have you tried using a `self.parent_view.modal.cancel.wait_displayed()` before executing the `click()`? Ping me and I can help you try to debug this.

Reply:
@john-dupuy 
Thanks :)
- I executed tests with ```self.parent_view.modal.cancel.wait_displayed()``` locally and it worked as expected. Hence I am updating with this solution. 
- I would like to see if it works on PRT or not.

Reply:
Update:
@john-dupuy It works with your solution but when I tried locally it gave me ```StaleElementReferenceException``` for some of those parametrized tests.
So are we going with this solution or you have any other sugestion?